src/components/TrungDay3CRUD/PageLocal/localread.js

```diff
@@ -0,0 +1,157 @@
+import { useState } from "react";
+import { Table, Button, Checkbox, Form } from "semantic-ui-react";
+
+const data = [
+  {
+    firstName: "Trung",
+    lastName: "Chi Nguyen",
+    checkbox: true,
+    id: 1,
+  },
+  {
+    firstName: "Toan",
+    lastName: "Nguyen",
+    checkbox: true,
+    id: 2,
+  },
+  {
+    firstName: "React",
+    lastName: "Crud",
+    checkbox: true,
+    id: 3,
+  },
+  {
+    firstName: "Nhan",
+    lastName: "Nguyen",
+    checkbox: true,
+    id: 4,
+  },
+  {
+    firstName: "Gandaf",
+    lastName: "iajsdf",
+    checkbox: true,
+    id: 5,
+  },
+];
+
+export default function ReadLocal() {
+  const [localData, setLocalData] = useState(data);
+  const [firstName, setFirstName] = useState("");
+  const [lastName, setLastName] = useState("");
+  const [checkbox, setCheckbox] = useState(false);
+  const [id, setId] = useState();
+
+  const setData = (data) => {
+    setId(data.id);
+    setFirstName(data.firstName);
+    setLastName(data.lastName);
+    setCheckbox(data.checkbox);
+  };
+
+  const onDelete = (id) => {
+    setLocalData(() => {
+      return localData.filter((item) => item.id !== id);
+    });
+    clearInputForm();
+  };
+
+  const handleSubmit = () => {
+    let newId = Date.now();
+    if (id) {
+      newId = id;
+      setLocalData((prev) => {
+        return prev.map((item) =>
+          item.id === id
+            ? {
+                firstName: firstName,
+                lastName: lastName,
+                checkbox: checkbox,
+                id: newId,
+              }
+            : item
+        );
+      });
+    } else {
+      setLocalData([
+        ...localData,
+        {
+          firstName: firstName,
+          lastName: lastName,
+          checkbox: checkbox,
+          id: newId,
+        },
+      ]);
+    }
+    clearInputForm();
+  };
+
+  const clearInputForm = () => {
+    setId("");
+    setFirstName("");
+    setLastName("");
+    setCheckbox(false);
+  };
+
+  return (
+    <div>
+      <Form className="create-form">
+        <Form.Field>
+          <label>First Name</label>
+          <input
+            value={firstName}
+            onChange={(e) => setFirstName(e.target.value)}
+            placeholder="First Name"
+          />
+        </Form.Field>
+        <Form.Field>
+          <label>Last Name</label>
+          <input
+            value={lastName}
+            onChange={(e) => setLastName(e.target.value)}
+            placeholder="Last Name"
+          />
+        </Form.Field>
+        <Form.Field>
+          <Checkbox
+            checked={checkbox}
+            label="I agree to the Terms and Conditions"
+            onChange={(e) => setCheckbox(!checkbox)}
+          />
+        </Form.Field>
+        <Button onClick={handleSubmit} type="submit">
```

Comment:
Event là onClick sao còn type="submit" ?

Reply:
Solved! pls check it out.